dsc_lib/src/discovery/command_discovery.rs

```diff
@@ -37,31 +58,75 @@ impl CommandDiscovery {
         }
     }
 
+    fn get_resource_path_setting() -> Result<ResourcePathSetting, DscError>
+    {
+        if let Ok(v) = get_setting("resource_path") {
+            // if there is a policy value defined - use it; otherwise use setting value
+            if v.policy != serde_json::Value::Null {
+                match serde_json::from_value::<ResourcePathSetting>(v.policy) {
+                    Ok(v) => {
+                        return Ok(v);
+                    },
+                    Err(e) => { return Err(DscError::Operation(format!("{e}"))); }
+                }
+            } else if v.setting != serde_json::Value::Null {
+                match serde_json::from_value::<ResourcePathSetting>(v.setting) {
+                    Ok(v) => {
+                        return Ok(v);
+                    },
+                    Err(e) => { return Err(DscError::Operation(format!("{e}"))); }
+                }
+            }
+        }
+
+        Err(DscError::Operation("Could not read 'resource_path' setting".to_string()))
```

Comment:
Probably better to have a new `DscError::Setting()` enum value instead of overloading `Operation`

Reply:
Updated.